### PR TITLE
chore(docs): Add examples to `Number` module

### PR DESCRIPTION
**Pull request by spotandjake**
This pr improves the documentation on the `Number` module, by adding examples to most functions. It also adds a missing `since` property to `Number.sign`.

Notes:
* I omitted examples on the functions that take a `Range` until we have proper syntax.
* Our trig functions are currently not precise enough for good examples, I think we might need to look into rewriting them #1478 had been an issue to make `Number.sin` unbounded, but I noticed during this pr that was not the actual problem, the problem is how we map the input range and also with our implementation.

### CODE REVIEW

File: stdlib/number.gr [hunk 1]
```diff
@@ -577,12 +705,16 @@ provide let sin = (radians: Number) => {
...
 }
 
+// TODO(#1478): Improve the examples with more accurate values
 /**
  * Computes the cosine of a number (in radians) using Chebyshev polynomials.
  *
  * @param radians: The input in radians
  * @returns The computed cosine
  *
+ * @example Number.cos(Number.pi) == -0.9999998749960952
```
Comment (alex-snezhko) [inline stdlib/number.gr:715]: I don't know if we should be showcasing that our `cos` implementation has flaws :laughing:

**Review by alex-snezhko — APPROVED** [review]

File: stdlib/number.gr (hunk 1 above)
Comment (spotandjake) [reply stdlib/number.gr:715]: I agree, I think the best option is to merge: #2032 as a temporary solution to our inaccuracies, and then fix this example.

**Comment by ospencer** [comment]
Let's make sure something akin to #2032 is merged before we land this.

**Comment by phated** [comment]
@spotandjake more conflicts (due to atan2?)

**Review by phated — APPROVED** [review]
LGTM but getting @ospencer or @alex-snezhko review

File: stdlib/number.gr [hunk 2]
```diff
@@ -337,6 +409,13 @@ provide let isInteger = (x: Number) => {
  * @param x: The number to check
  * @returns `true` if the value is a non-integer rational number or `false` otherwise
  *
+ * @example Number.isRational(1/2)
```
Comment (ospencer) [inline stdlib/number.gr:412]: No == true here?

File: stdlib/number.gr [hunk 3]
```diff
@@ -351,6 +430,14 @@ provide let isRational = (x: Number) => {
  * @param x: The number to check
  * @returns `true` if the value is finite or `false` otherwise
  *
+ * @example Number.isFinite(1/2)
+ * @example Number.isFinite(0.5)
+ * @example Number.isFinite(1.0)
+ * @example Number.isFinite(1)
```
Comment (ospencer) [inline stdlib/number.gr:436]: And these?

Comment (ospencer) [reply stdlib/number.gr:436]: Or is this our style? Just feels a little weird since it's not an assertion.

Comment (phated) [reply stdlib/number.gr:436]: I haven't been paying that close of attention. I know jake was doing assert in some cases.

**Review by ospencer — APPROVED** [review]

File: stdlib/number.gr (hunk 3 above)
Comment (spotandjake) [reply stdlib/number.gr:436]: We have been using `assert` semantics for single line matches throughout the stdlib.

Comment (phated) [reply stdlib/number.gr:436]: @spotandjake why don't these contain `assert` then?

**Pull request merged**